### PR TITLE
leveldb.cmake: add URL_HASH to improve security

### PR DESCRIPTION
**Pull request by dconeybe**

### CODE REVIEW

File: cmake/external/leveldb.cmake [hunk 1]
```diff
@@ -26,6 +26,7 @@ ExternalProject_Add(
   DOWNLOAD_DIR ${FIREBASE_DOWNLOAD_DIR}
   DOWNLOAD_NAME leveldb-${version}.tar.gz
   URL https://github.com/google/leveldb/archive/${version}.tar.gz
+  URL_HASH SHA256=9a37f8a6174f09bd622bc723b55881dc541cd50747cbd08831c2a82d620f6d76
```
Comment (ehsannas) [inline cmake/external/leveldb.cmake:29]: is this the SHA256 of `https://github.com/google/leveldb/archive/1.23.tar.gz` ?

I got `a1c64afd13a4a4e3cc5a4d8e04d7b2ec4c95208bd2b51b29e583598da5675b67` 😄

Comment (dconeybe) [reply cmake/external/leveldb.cmake:29]: Huh, can you try again? I got `9a37f8a6174f09bd622bc723b55881dc541cd50747cbd08831c2a82d620f6d76`

Here is what I saw (with some noise removed):

```
$ wget https://github.com/google/leveldb/archive/1.23.tar.gz
HTTP request sent, awaiting response... 200 OK
Length: 242925 (237K) [application/x-gzip]
Saving to: ‘1.23.tar.gz’

$ sha256sum 1.23.tar.gz
9a37f8a6174f09bd622bc723b55881dc541cd50747cbd08831c2a82d620f6d76  1.23.tar.gz
```

Comment (ehsannas) [reply cmake/external/leveldb.cmake:29]: oops 🤦‍♂️  sorry. it's correct.

**Review by ehsannas — APPROVED** [review]

File: cmake/external/leveldb.cmake (hunk 1 above)
Comment (dconeybe) [reply cmake/external/leveldb.cmake:29]: no problem. that is one thorough review! thanks!

**Pull request merged**